src/main/java/dao/impl/WareHouseDaoImpl.java

```diff
@@ -0,0 +1,55 @@
+package dao.impl;
+
+import dao.WareHouseDao;
+import db.WareHouse;
+import java.util.Arrays;
+
+public class WareHouseDaoImpl implements WareHouseDao {
+    private static final int ZERO = 0;
+
+    @Override
+    public int getStoredQuantity(String fruitName) {
+        if (isFruitPresence(fruitName)) {
+            String[] fruitLot = WareHouse.STORED_FRUITS
+                    .stream()
+                    .filter(array -> array[0].equals(fruitName))
+                    .findFirst().get();
```

Comment:
Improve performance from O(N) to O(1) by using hashmap for storage, where fruitName String is key